assets/js/custom/Translation.js

```diff
@@ -1,40 +1,134 @@
 /* eslint-env jquery */
 
 // eslint-disable-next-line no-unused-vars
-function Translation (hasDescription, hasCredit) {
-  const elementsToTranslate = [document.getElementById('name')]
-
-  if (hasDescription) {
-    elementsToTranslate.push(document.getElementById('description'))
-  }
-  if (hasCredit) {
-    elementsToTranslate.push(document.getElementById('credits'))
+function Translation (translatedByLine) {
+  const providerMap = {
+    itranslate: 'iTranslate'
   }
+  let languageMap = {}
+  let translatedByLineMap = {}
+  let targetLanguage = null
+
+  $(document).ready(function () {
+    setTargetLanguage()
+    getLanguageMap()
+    splitTranslatedByLine()
+  })
+
+  $(document).on('click', '.comment-translation-button', function () {
+    const commentId = $(this).attr('id').substring('comment-translation-button-'.length)
 
-  translation(
-    document.getElementById('translate-program'),
-    elementsToTranslate,
-    document.documentElement.lang
-  )
+    $(this).hide()
+
+    if (isTranslationNotAvailable(commentId)) {
+      $('#comment-translation-loading-spinner-' + commentId).show()
+      translateComment(commentId)
+    } else {
+      openTranslatedComment(commentId)
+    }
+  })
 
-  $('.comment-translate-button').each(function () {
-    const commentId = $(this).attr('id').substring('comment-translate-button-'.length)
-    translation(this, document.getElementById('comment-text-' + commentId), document.documentElement.lang)
+  $(document).on('click', '.remove-comment-translation-button', function () {
+    const commentId = $(this).attr('id').substring('remove-comment-translation-button-'.length)
+    $(this).hide()
+    $('#comment-translation-button-' + commentId).show()
+    $('#comment-translation-wrapper-' + commentId).slideUp()
+    $('#comment-text-wrapper-' + commentId).slideDown()
   })
 
-  function translation (buttonElement, textElements, targetLang) {
-    let text = ''
-    if (Array.isArray(textElements)) {
-      const array = []
-      for (let i = 0; i < textElements.length; i++) {
-        array.push(textElements[i].innerText)
+  function setTargetLanguage () {
+    targetLanguage = document.documentElement.lang
+    const decodedCookie = decodeURIComponent(document.cookie).split(';')
+    for (let i = 0; i < decodedCookie.length; i++) {
+      if (decodedCookie[i].includes('hl=') && decodedCookie[i].length < 10) {

```

Comment:
Can you explain a bit why the `length < 10` is needed? this might not be obvious

Reply:
It's just an extra check to make sure that the cookie doesn't include anything else when it is split. For example, in the case where the cookie is "extratexthl=value; hl=en;", it won't set the target language to "value" instead of "en"

Reply:
that is very hacky. you might want to look at this https://www.w3resource.com/javascript-exercises/fundamental/javascript-fundamental-exercise-171.php

Reply:
That looks like a lot better solution, I will update it